src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1487,9 +1471,8 @@ internal void FixupRowDescription(RowDescriptionMessage rowDescription, bool isF
             }
         }
 
-        void LogCommand()
+        void LogCommand(NpgsqlConnector connector)
         {
-            var connector = _connection!.Connector!;
             var sb = new StringBuilder();
```

Comment:
I'd say this entire method should be in a try catch.

Reply:
There should never be an exception here... Do you mean like a defensive try/catch to guard against more bugs we make?

Reply:
Yeah maybe not but I see quite a few grabs into other instance state, I guess it could be ok if it's under test ;)

Reply:
I'm not sure it's really under test :fearful: (the logging system is very static/global so it's a bit tricky). But I don't like the idea of going into programmer-defensive try/catch, cause bugs can occur anywhere

Reply:
For logging this is usually the right approach, as its blast radius should be contained under weird cases. But anyway know I don't entirely disagree here.

Reply:
Ideally we'd just add test coverage here... A bug like what we recently found (with the connector) is indeed annoying, but it's also very fail-fast and easy to fix, so the problem is problem not too huge.